cli/azd/internal/runcontext/first_run.go

```diff
@@ -0,0 +1,63 @@
+package runcontext
+
+import (
+	"errors"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+// firstRunFileName is the name of the file created in the azd configuration
+// directory after the first run of the CLI. It's presence is used to determine
+// if this is the first run of the CLI.
+const cFirstRunFileName = "first-run"
+
+func IsFirstRun() bool {
+	firstRunFilePath, err := getFirstRunFilePath()
+	if err != nil {
+		log.Printf("failed to get first run file path: %v", err)
+		// If the first run file path can't be resolved assume it is the first
+		// run and return true
+		return true
```

Comment:
I worry a little when we create application-level features like `runcontext`, and we make decisions like: `IsFirstRun()` should return `true` when unhandled errors occur. It may be better to default to `true` or `false` depending on feature usage. This isn't a big deal, but it's why my initial design pass on the requirements is to create a `cloudshell-telemetry-notice` that is tied directly to the feature usage.

Reply:
After chatting with Wei: This is obviously a case of YAGNI. This logic belongs in the telemetry package until it's actually needed elsewhere. 

The logic in `main.go` should be reduced down to some kind of call to telemetry like `telemetry.ShouldDisplayFirstRunMessage` which would check things like whether we're running in Cloud Shell, whether there's been a "first-run" file created, etc.